context_chat_backend/chain/ingest/injest.py

```diff
@@ -122,17 +122,16 @@ def _process_sources(
 	vectordb: BaseVectorDB,
 	config: TConfig,
 	sources: list,
-	result: dict[multiprocessing.Event],
+	result: dict[str, multiprocessing.Event],  # pyright: ignore[reportInvalidTypeForm]
 	embedding_taskqueue: multiprocessing.Queue,
-):
+) -> None:
 	filtered_sources = _filter_sources(sources[0].get('userId'), vectordb, sources)
 
 	if len(filtered_sources) == 0:
 		# no new sources to embed
 		print('Filtered all sources, nothing to embed', flush=True)
-		result['success'].set()
 		result['done'].set()
-		return True
+		result['success'].set()
```

Comment:
This is the wrong way around, I think. Controller code waits for done. If success is set after done, we have a race condition.

Reply:
ah you're right, this must be the deadlock in the integration test